### PR TITLE
CDPS-786: Adding logging and telemetry to sync service

### PR DESCRIPTION
**Pull request by brightonsbox**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonperson/service/PhysicalAttributesSyncService.kt [hunk 1]
```diff
@@ -96,4 +107,18 @@ class PhysicalAttributesSyncService(
     fieldHistory.last { it.field == HEIGHT },
     fieldHistory.last { it.field == WEIGHT },
   ).map { it.fieldHistoryId }
+
+  private fun trackSyncEvent(prisonerNumber: String, fieldHistoryIds: List<Long>) {
+    telemetryClient.trackEvent(
+      "prison-person-api-physical-attributes-synced",
```
Comment (davidatmoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonperson/service/PhysicalAttributesSyncService.kt:113]: Is this ok to be the same for both "normal" sync and historical? Or should it be passed in as a param?

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonperson/service/PhysicalAttributesSyncServiceTest.kt [hunk 2]
```diff
@@ -88,6 +96,12 @@ class PhysicalAttributesSyncServiceTest {
           HistoryComparison(value = PRISONER_WEIGHT, createdAt = NOW, createdBy = USER1, appliesFrom = NOW, appliesTo = null, source = NOMIS),
         )
       }
+
+      verify(telemetryClient).trackEvent(
```
Comment (davidatmoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonperson/service/PhysicalAttributesSyncServiceTest.kt:100]: Might need changes if other comment accepted. Same for other instances of `verify`

**Review by davidatmoj — APPROVED** [review]
Added some minor comments, but happy for them to be disregarded if it makes sense.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonperson/service/PhysicalAttributesSyncService.kt (hunk 1 above)
Comment (brightonsbox) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/prisonperson/service/PhysicalAttributesSyncService.kt:113]: I wasn't particularly bothered about distinguishing them really, we still have different log messages if we really wanted to do a count of historic edits I suppose

**Pull request merged**